src/components/CurrencyInputPanel/CurrencyInputPanel.component.tsx

```diff
@@ -0,0 +1,199 @@
+            {!hideInput && (
+              <>
+                <NumericalInput
+                  className="token-amount-input"
+                  value={value}
+                  onFocus={handleFocus}
+                  onBlur={handleBlur}
+                  onUserInput={onUserInput}
+                  disabled={disabled}
+                  data-testid={'transaction-value-input'}
+                />
+              </>
+            )}
+            <CurrencySelect
+              selected={!!(currency || pair)}
+              className="open-currency-select-button"
+              disableCurrencySelect={disableCurrencySelect}
+              onClick={() => {
+                if (!isLoading && !disableCurrencySelect) {
+                  setIsOpen(true)
+                }
+              }}
+              disabled={(isLoading || disableCurrencySelect) && !currency}
+            >
+              <Aligner>
+                {isLoading ? (
+                  <Loader style={{ width: '97.25px' }} />
+                ) : (
+                  <>
+                    {pair ? (
+                      <DoubleCurrencyLogo marginRight={4} currency0={pair.token0} currency1={pair.token1} size={20} />
+                    ) : currency ? (
+                      <CurrencyLogo
+                        size="20px"
+                        currency={currency}
+                        chainIdOverride={chainIdOverride}
+                        currencyWrapperSource={currencyWrapperSource}
+                      />
+                    ) : null}
+                    {pair ? (
+                      <StyledTokenName className="pair-name-container">
+                        {pair?.token0.symbol}/{pair?.token1.symbol}
+                      </StyledTokenName>
+                    ) : (
+                      <StyledTokenName className="token-symbol-container" active={Boolean(currency && currency.symbol)}>
+                        {(currency && currency.symbol && currency.symbol.length > 20
+                          ? currency.symbol.slice(0, 4) +
+                            '...' +
+                            currency.symbol.slice(currency.symbol.length - 5, currency.symbol.length)
+                          : currency?.symbol) || <div data-testid="select-token-button"> {t('select Token')}</div>}
+                      </StyledTokenName>
+                    )}
+                    {!disableCurrencySelect && (pair || currency) && <StyledDropDown selected={!!currency} />}
+                  </>
+                )}
+              </Aligner>
+            </CurrencySelect>
+          </InputRow>
+          <FiatRow>
+            <RowBetween>
+              {fiatValue && (
+                <FiatValueDetails
+                  fiatValue={fiatValue?.toFixed(2, { groupSeparator: ',' })}
+                  priceImpact={priceImpact}
+                  isFallback={isFallbackFiatValue}
+                />
+              )}
```

Comment:
May be pass in fiatValue and handle the whole logic of `fiatValue` inside the component and if its undefined or null return null
```suggestion
             
                <FiatValueDetails
                  fiatValue={fiatValue}
                  priceImpact={priceImpact}
                  isFallback={isFallbackFiatValue}
                />
        
```

Reply:
Fixed. You can take a look again :)